### DISCUSSION TITLE
about query result distances

### DISCUSSION THREAD

**Discussion by gaoyuan98** [milvus-io/milvus#26290] [Q&A and General discussion] [answered]
Using L2 to query, the distance value returned is sorted from low to high when the topk is set to 3. I would like to ask if the lower the distance value, the higher the matching degree
![image](https://github.com/milvus-io/milvus/assets/44916277/b45d1988-d61b-436b-a020-ab550d662cb0)

**Comment by yhmo** [comment, marked as answer]
It depends on the metric type: https://milvus.io/docs/metric.md

When you creating an index by create_index() ,there is  parameter "metric_type":
```
    index_param = {
        "index_type": "IVF_FLAT",
        "params": {"nlist": 1024},
        "metric_type": "L2"}
    collection.create_index(filed_name, index_param)
```

The search() interface also has a parameter "metric_type", 
```
search_param = {
        "data": search_vectors,
        "anns_field": vector_field,
        "param": {"metric_type": "L2", "params": {"nprobe": 10}},
        "limit": 10}
    results = collection.search(**search_param)
```

We require the "metric_type" must be the same for create_index() and search().
L2 is Euclidean distance, a smaller value means more similar. Zero value means the two vectors are equal.
IP is Inner product, this kind of metric type requires all the vectors are normalized(length to be 1), IP distance value is between [0, 1]. If the distance is 1, the two vectors are equal. The value is closer to 1, the two vectors are more similar.

**Reply by echalabiyev** [reply]
Hello,
So does it mean if I have chosen "COSINE" as a metric_type, then the distance actully will show how close the results are, so higher that value better the result?

**Reply by yhmo** [reply]
The cosine similarity is always in the interval [-1, 1]. For example, two proportional vectors have a cosine similarity of 1, two orthogonal vectors have a similarity of 0, and two opposite vectors have a similarity of -1. The larger the cosine, the smaller the angle between the two vectors, indicating that these two vectors are more similar to each other.
